Mono - Mangled Results

emiaj opened the pull request:
This commit fixes a problem when using Spark under Mono - Linux.

The issue can be reproduced using the FubuMVC.HelloSpark sample from the FubuMVC project.

What it does is to prevent that SpoolPage gets released  multiple times.

I think the problem occours when a buffer obtained from the Allocator gets overriden when a child SpoolPage is released before its parent when running the finalizers:

SpoolPageA -> SpoolPageB -> SpoolPageC

SpoolPageB.Release() also calls to SpoolPageC.Release()

Later SpoolPageA is released, causing once again to release B and C, which makes the content that is currently being written into one of those writers to get cleared, thus, the mangled results.

With this flag we prevent that SpoolPage instances get released more than once (because of the _released flag).

RobertTheGrey committed:
Party with Path.DirectorySeparatorChar - all tests passing

RobertTheGrey committed:
Party with Path.DirectorySeparatorChar - all tests passing

RobertTheGrey committed:
Party with Path.DirectorySeparatorChar - all tests passing

JoshClose committed:
Removed generic class constraint for SparkView.

JoshClose committed:
Removed generic class constraint for SparkView.

RobertTheGrey committed:
Merge branch 'mvc2-net35' into mvc2-net4

RobertTheGrey committed:
Merge branch 'mvc2-net4'

RobertTheGrey committed:
Preparing for release 1.5.1

RobertTheGrey committed:
Preparing for release 1.5.1

RobertTheGrey committed:
Merge branch 'mvc2-net4'

RobertTheGrey committed:
Preparing for release 1.5.1

RobertTheGrey committed:
Added Futures packages

RobertTheGrey committed:
Fixed WebActivator Conventions and references http://blog.davidebbo.com/2011/02/appstart-folder-convention-for-nuget.html

RobertTheGrey committed:
Merge branch 'mvc2-net35' into mvc2-net4

RobertTheGrey committed:
Fixed WebActivator Conventions and references http://blog.davidebbo.com/2011/02/appstart-folder-convention-for-nuget.html

RobertTheGrey committed:
Fixed WebActivator Conventions and references http://blog.davidebbo.com/2011/02/appstart-folder-convention-for-nuget.html

schotime committed:
Fix whitespace issues and provide extra tests

ahjohannessen committed:
Mono/Linux adjustments.
Rob: Please take a look at the comments/changes in:

- VisualBasicViewCompilerTester.cs
- SubViewFolderTester.cs

ahjohannessen committed:
Additional Mono/Linux adjustments.

ahjohannessen committed:
Additional Mono/Linux adjustments.

ahjohannessen committed:
Additional Mono/Linux adjustments.

ahjohannessen committed:
Upgrading NUnit to v2.5.9 + cleaning up in libs.
Adjusted test code to new version of nunit -> removing usage of obsolete methods and namespace fixes.

RobertTheGrey committed:
Merge branch 'mono-spike' of https://github.com/ahjohannessen/spark into ahjohannessen-mono-spike

RobertTheGrey committed:
Merge branch 'mvc2-net35' into mvc2-net4

RobertTheGrey committed:
Mono/Linux adjustments.

RobertTheGrey committed:
Mono/Linux adjustments.

RobertTheGrey committed:
Merge branch 'schotime-mvc2-net35'

RobertTheGrey committed:
Merge branch 'mvc2-net4' into mvc3-net4

RobertTheGrey committed:
Merge branch 'mvc2-net35' into mvc3-net4

RobertTheGrey committed:
Merge branch 'mvc2-net35' into mvc2-net4

RobertTheGrey committed:
Merge branch 'mvc2-net4' into mvc3-net4

RobertTheGrey committed:
Path.Combine hacks to get mono changes working on .Net 3.5

RobertTheGrey committed:
Nuget package update

RobertTheGrey committed:
Nuget package update

RobertTheGrey committed:
Nuget package update

RobertTheGrey committed:
Nuget package update

RobertTheGrey committed:
Merge branch 'mvc2-net4' into mvc3-net4

RobertTheGrey committed:
Nuget package update

ahjohannessen committed:
Mono/Linux adjustments : BindingExecutionTester.cs

ahjohannessen committed:
Temporary Fix for Mono/Linux - Latest version of Mono (2.10.1) has a bug in its TextReader implementation.

Makes following tests in SpoolReaderTester pass:

- Empty_and_null_writes_are_acceptable
- Multiple_writes_are_combined
- StringReader_should_return_written_data

RobertTheGrey committed:
Fix for MVC areas and removed some ignored tests
(cherry picked from commit 74438d27edbd43d39a245f3e010ceb26ce88a78d)

RobertTheGrey committed:
Unit test fixes due to NUnit upgrade and version update

RobertTheGrey committed:
Unit test fixes due to NUnit upgrade and version update

RobertTheGrey committed:
Merge branch 'mvc2-net4' into mvc3-net4

RobertTheGrey committed:
Nuget update

Commit:
Made the SpoolPage to safely release its resources.

RobertTheGrey commented:
Hi Jaime,

You'll have to resubmit the pull request to the main repo master branch - this branch is loudej's private fork which I have no control over.

emiaj commented:
Hi Robert,
Done!
https://github.com/RobertTheGrey/spark/pull/4
Regards,
Jaime.

ahjohannessen committed:
Mono/Linux adjustments + improvements on making the core parts work out-of-the-box in a MonoDevelop environment:

- Excluded c# compiler directive in Spark.Tests/App.config as this is causing troubles in Mono/Linux.
- Made an exclusion on all Mono tests in VisualBasicViewTester.cs.
- Fixed up excluded test in VisualBasicViewCompilerTester.cs to use NUnit attribute.

ahjohannessen committed:
Mono/Linux : Removed unused references to System.Web.Mobile and System.Web.Entity - These don't exist on Mono and make the out-of-the-box experience a bit rougher with a clean checkout.

emiaj committed:
Added unit test to reproduce the bug on SpoolPage (Release method called on _next members before its owner).

RobertTheGrey committed:
Update to version 1.5.1.3

RobertTheGrey committed:
Merge branch 'mvc2-net35' into mvc2-net4

RobertTheGrey committed:
Merge branch 'mvc2-net4' into mvc3-net4

RobertTheGrey committed:
Nuget update

RobertTheGrey committed:
Merged pull request #5 from ahjohannessen/master.

More Mono/Linux adjustments

mkmurray committed:
Add 'ignore' tag that skips all Spark parsing of inner content.

mkmurray committed:
Bug fix for 'ignore' tag to allow other closing markup tags within its scope

RobertTheGrey committed:
Merge pull request #17 from mkmurray/master.

Add 'ignore' tag that skips all Spark parsing of inner content

RobertTheGrey committed:
Updated Spark version

RobertTheGrey committed:
Updated NuGet packages

RobertTheGrey committed:
Merge branch 'master' of github.com:RobertTheGrey/spark

emiaj committed:
First spike on getting a hold on the view directory path when resolving markup binders in the provider, subject to reformulation.

emiaj committed:
Made the view loader pass the view path instead of the directory path to binding providers.

emiaj committed:
Added BindingRequest param object.

emiaj committed:
Removed IEqualatable members from BindingRequest.
